opentracing-tracerresolver/src/main/java/io/opentracing/contrib/tracerresolver/TracerResolver.java

```diff
@@ -44,16 +44,33 @@
      * Resolves the {@link Tracer} implementation.
      *
      * @return The resolved Tracer or {@code null} if none was resolved.
+     * @deprecated Tracers are encouraged to use the {@link TracerFactory} interface to provide tracers
      */
+    @Deprecated
     protected abstract Tracer resolve();
 
     /**
-     * Detects all {@link TracerResolver} service implementations and attempts to resolve a {@link Tracer}.
+     * Attempts to resolve a Tracer via {@link ServiceLoader} using a variety of mechanisms, from the most recommended
+     * to the least recommended:
+     * <p>
+     * <ul>
+     *     <li>based on the available {@link TracerFactory}</li>
+     *     <li>based on subclasses of {@link TracerResolver}</li>
+     *     <li>based on classes of {@link Tracer}</li>
+     * </ul>
+     *
+     * <p>
+     * Whenever a Tracer can be resolved by any of the methods above, the resolution stops. It means that if a Factory
+     * is found, no Resolvers are attempted to be loaded.
+     *
      * <p>
      * If a {@code GlobalTracer} has been previously registered, it will be returned before attempting to resolve
      * a {@linkplain Tracer} on our own.
+     *
      * <p>
-     * If there are more than one resolver, the first non-<code>null</code> resolved tracer is returned.
+     * If more than one {@link TracerFactory} or {@link TracerResolver} is found, the one with the highest priority is
+     * returned. Note that if a {@link TracerResolver} has a higher priority as all available {@link TracerFactory},
+     * the factory still wins.
```

Comment:
>  Note that if a {@link TracerResolver} has a higher priority as all available {@link TracerFactory}, the factory still wins.

nit: _higher than_  
:+1: I like this addition, makes the overall priority abundantly clear.